### ISSUE TITLE
how to use vscode's html for <template> format?

### ISSUE THREAD

**Issue by transtone** [vuejs/vetur#488]
- [x] I have searched through existing issues
- [x] I have read through docs at https://vuejs.github.io/vetur/
- [x] I have read FAQ: https://github.com/vuejs/vetur/wiki/FAQ

## Info

- Platform: macOS
- Vetur version: 0.10.1
- VSCode version: 1.17.2

## Problem

when save   comment in <template>, the code auto add 4 whitespace in all comment line everytime. 

## Reproducible Case

```
<!--
  Important. Please provide clear steps for reproducing the problem.
  Otherwise we can't help you and your issue might be closed.
  For example, fork https://github.com/octref/veturpack and modify it to reproduce the error,
  then push your changes to GitHub and send us the link.
-->
```

``` 
<!--
                             Important. Please provide clear steps for reproducing the problem.
                            Otherwise we can't help you and your issue might be closed.
                            For example, fork https://github.com/octref/veturpack and modify it to reproduce the error,
                            then push your changes to GitHub and send us the link.
-->
```

use `html` for .vue files works fine. but `vue` for .vue files got this problem.

I kown it must js-beautify's problem, but I can't find a issue about this, and vscode's html(use js-beautify?) works fine. Is there any template formatter tools to replace js-beautify-html?

**Comment by octref** [comment]
![image](https://user-images.githubusercontent.com/4033249/31805509-1d554cbe-b516-11e7-884d-5c8a775fbf72.png)

If you can fix broken thign caalled js-beautify I'll buy you a beer.
The defualt setting doesn't work. I tried all possible combination of the settings I can imagine it still doesn't work. For the sake of sanity if there is one more js-beautify related issue I'll remove it immediately.

**Comment by transtone** [comment]
vscode's html tool not have this bug:
https://github.com/Microsoft/vscode-html-languageservice/blob/master/src/services/htmlFormatter.ts

**Comment by HerringtonDarkholme** [comment]
In neither html nor vue can I reproduce this problem.

**Comment by HerringtonDarkholme** [comment]
@octref  This is caused by rewrapping code in indentation.

https://github.com/vuejs/vetur/blob/master/server/src/modes/template/services/htmlFormat.ts#L16

js-beautify won't deindent code in comment. So effectively it reindent. 

Sadly I don't have time to fix this. :/

**Comment by octref** [comment]
> js-beautify won't deindent code in comment.

Yeah, but it indents the first line of comment, and not the rest, which I think is not reasonable as a beautifier. It should either leave all comments alone or indent all comments together.

![image](https://user-images.githubusercontent.com/4033249/32630616-2604f796-c552-11e7-9873-fae1e135c8fd.png)

**Comment by transtone** [comment]
```In neither html nor vue can I reproduce this problem. ```
Do you set `  "editor.formatOnType": true` ?

I clear all vscode settings, 
just install `vetur`, 
and set `"editor.formatOnType": true`,
save file with `meta-s`

this is what happend:

![untitled](https://user-images.githubusercontent.com/46688/32634359-33cf4f04-c5e6-11e7-8e59-ab98fc9efcef.gif)

**Comment by transtone** [comment]
@HerringtonDarkholme  can you reproduce this problem?

**Comment by HerringtonDarkholme** [comment]
I can reproduce but I will not fix this.

**Comment by transtone** [comment]
It fixed in vetur-0.11.3, thanks!
